src/dbnode/storage/shard.go

```diff
@@ -1687,8 +1687,10 @@ func (s *dbShard) FetchBlocksMetadataV2(
 			pos.metadataIdx = int(flushedPhase.CurrBlockEntryIdx)
 		}
 
+		// TODO(juchan): actually get the volume
+		vol := 0
 		// Open a reader at this position, potentially from cache
-		reader, err := s.namespaceReaderMgr.get(s.shard, blockStart, pos)
+		reader, err := s.namespaceReaderMgr.get(s.shard, blockStart, vol, pos)
```

Comment:
hmm....we may need to make this thing listen for lease changes...I'll dig into it more in my P.R

Reply:
Filed an issue to track: https://github.com/m3db/m3/issues/1732

I'll remove the volume arg for now.